source/flux/parts/courses.js

```diff
@@ -68,23 +69,70 @@ export function updateCourseData(): UpdateCourseDataActionType {
 	}
 }
 
+const LOAD_RECENT_SEARCHES = 'courses/LOAD_RECENT_SEARCHES'
+
+type LoadRecentSearchesAction = {|
+	type: 'courses/LOAD_RECENT_SEARCHES',
+	payload: string[],
+|}
+export async function loadRecentSearches(): Promise<LoadRecentSearchesAction> {
+	const recentSearches = await storage.getRecentSearches()
+	return {type: LOAD_RECENT_SEARCHES, payload: recentSearches}
+}
+
+const UPDATE_RECENT_SEARCHES = 'courses/UPDATE_RECENT_SEARCHES'
+
+type UpdateRecentSearchesAction = {|
+	type: 'courses/UPDATE_RECENT_SEARCHES',
+	payload: string[],
+|}
+export function updateRecentSearches(
+	query: string,
+): ThunkAction<UpdateRecentSearchesAction> {
+	return (dispatch, getState) => {
+		const state = getState()
+
+		const oldRecentSearches = state.courses ? state.courses.recentSearches : []
+		if (oldRecentSearches.includes(query)) {
```

Comment:
Should we lowercase these while comparing them, to avoid storing ones that only vary in case?

Reply:
I thought about that...just not long enough to actually do something about it! But yeah it's probably a good call.